pilot/pkg/bootstrap/server.go

```diff
@@ -443,7 +449,47 @@ func (c *mockController) Run(<-chan struct{}) {}
 
 // initConfigController creates the config controller in the pilotConfig.
 func (s *Server) initConfigController(args *PilotArgs) error {
-	if args.Config.Controller != nil {
+	mcpServerAddrs := args.MCPServerAddrs
+	if len(mcpServerAddrs) > 0 {
+		// TODO: log random uuid and assign to this pilot node
+		clientNodeID := "some-id"
+		supportedTypes := make([]string, len(model.IstioConfigTypes))
+		for _, model := range model.IstioConfigTypes {
+			supportedTypes = append(supportedTypes, model.MessageName)
+		}
+		log.Infof("mcp client initialized with supported types: %+v\n", supportedTypes)
+		logger := log.RegisterScope("coredatamodel", "mcp controller debugging", 0)
+		mcpController := coredatamodel.NewController(logger)
```

Comment:
What is this ? Passing logger to components ?

Reply:
That was to be able to inject a fake logger for testing purposes, but I agree since it is the only argument to that constructor, it seems a little pointless to assert on logs.